src/test/java/com/problem1/myproject/service/implementation/CoinServiceTest.java

```diff
@@ -0,0 +1,88 @@
+package com.problem1.myproject.service.implementation;
+
+import com.problem1.myproject.exceptions.ObjectNotFoundException;
+import com.problem1.myproject.model.Coin;
+import com.problem1.myproject.repository.CoinRepository;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Disabled;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Mock;
+
+import org.mockito.MockitoAnnotations;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import java.util.List;
+
+import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
+import static org.assertj.core.api.AssertionsForClassTypes.assertThatThrownBy;
+import static org.mockito.Mockito.verify;
+
+@ExtendWith(MockitoExtension.class)
+class CoinServiceTest {
+
+    @Mock private CoinRepository coinRepo;
+    private CoinService underTest;
+
+
+    @BeforeEach
+    void setUp(){
```

Comment:
@RobertRodeapps this in not good practice. Please use the InjectMocks annotation when declaring coinService